server/github_pr.go

```diff
@@ -61,18 +60,14 @@ func (d *EventData) HasUpdateBranch() bool {
 	return strings.Contains(strings.TrimSpace(*d.Comment.Body), "/update-branch")
 }
 
-func PullRequestEventFromJSON(data io.Reader) (*PullRequestEvent, error) {
+func PullRequestEventFromJSON(data io.Reader) *PullRequestEvent {
 	decoder := json.NewDecoder(data)
 	var event PullRequestEvent
 	if err := decoder.Decode(&event); err != nil {
-		return nil, err
-	}
-
-	if event.Issue == nil {
```

Comment:
I don't see why removing this would have any effect. If `event.Isssue` is nil then the `err` would be non-nil so `if err == nil && event.PRNumber != 0 {` block would never get executed.

Reply:
Not sure I understand your point completely. If event.Issue is nil, we were returning an error previously, which prevented the `if err == nil && event.PRNumber != 0 {` block from getting executed. Now it will.

Perhaps I am missing something?

Reply:
I have the impression if `event.Issue` is not nil then `event.PRNumber` is `0`. Isn't this the case?

Reply:
Not really. It's not mutually exclusive. The `Issue` field might be there for both scenarios. But the `PRNumber` will only be there if it's a PR.

It's super confusing, and that's why we are moving to separate endpoints.

Reply:
Oh I see, it makes sense then!